### ISSUE TITLE
[RLLIB] [Improvement] GPU-based SampleCollector and VectorEnv for massive parallelization

### ISSUE THREAD

**Issue by AlessandroZavoli** [ray-project/ray#23863]
### Description

Similarly to NVIDIA's approach for massively parallelization in Isaac Sim/Gym, I would suggest developing a pair of `GPUSampleCollector` and `GPUVectorEnv` for RLLIB based on PyTorch.
The idea is that all computations in the GPUVectorEnv are done in the GPU, thus allowing for vectorization/parallelization over GPU cores.
My guess is that both the vectorized environment and the Neural network must run on the GPU. In this way, data never go back to the host (CPU) and we avoid bottlenecks.

### Use case

The goal is to have a way to run a massive number of simple robotic/control environments in parallel, using the GPU to speed up the collection of sample "state-action-reward" tuples by various orders of magnitude with respect to traditional, CPU-based, environment

**Comment by P90-RushB** [comment]
Hope rllib can support the training of Isaac gym

**Comment by AlessandroZavoli** [comment]
Anyone interested in?

**Comment by AlessandroZavoli** [comment]
Keep it relevant

**Comment by karlkurzer** [comment]
Hey ray team, is this something that you still consider, or has this already been implemented? The docs are not 100% clear to me.
